### DISCUSSION TITLE
FB_Executorへのjob_event_reporterプロパティ設定タイミング

### DISCUSSION THREAD

**Discussion by makitagawa243** [Beckhoff-JP/PLC_JobManagementFramework#10] [General]
現状のMAINプログラムでは、最上位executorのjob_event_reporterプロパティが毎スキャン代入されています。
ジョブ構築後に1回Setするのが適切ではないかと考えたのですが、意図した実装でしょうか？
（設定漏れがなくなるメリットはあると思います）

Queueジョブの場合は毎回代入し直さないと新しく追加したExecutorに伝播しないので
毎スキャン代入することで現状は問題なく動いている認識です。
厳密には、add_futureした次のスキャンでjob_event_reporterが伝播するので隙間は存在するように思います。

job_event_reporterのSetタイミングをいつでもOKに、かつ隙間を作らないよう
job_event_reporter.Set内でのコンテナ内伝播に加えて、コンテナに対してadd_futureした際にSetするようにすれば
どう書いても意図通り設定できるのではないでしょうか。
ただし、現状はコンテナ自身がjob_event_reporterを持っておらず、FB_Executor.job_event_reporterのGetは未実装のため
合わせて検討は必要です。

**Comment by ta-ichihashi** [comment]
さすがですね。なかなか思いつきませんでしたがその通りです。add,_futureの後でも先でも確実に伝搬てきるようにしたいですね。JobContainerの親のExecutorオブジェクトが取れるので、add_future()のときに親からreporterオブジェクトをゲットして伝搬させますね。

**Comment by makitagawa243** [comment]
#13 にて解決済、クローズします。
